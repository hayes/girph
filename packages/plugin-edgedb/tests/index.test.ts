import schema from './example/schema';
import { db as edgedb } from './example/db';
import { execute, printSchema } from 'graphql';
import { gql } from 'graphql-tag';

let queries = [];

describe('edgedb', () => {
  afterEach(() => {
    queries = [];
  });

  it('generates schema', () => {
    const graphqlSchema = printSchema(schema);
    expect(graphqlSchema).toMatchSnapshot();
  });

  it('queries for single item', async () => {
    const query = gql`
      query {
        me {
          id
        }
      }
    `;

    const result = await execute({
      schema,
      document: query,
      contextValue: { user: { id: 'a04bf8b8-1bfd-11ed-93f8-836b78753212' } },
    });

    expect(result).toMatchInlineSnapshot(`
      Object {
        "data": Object {
          "me": Object {
            "id": "a04bf8b8-1bfd-11ed-93f8-836b78753212",
          },
        },
      }
    `);
  });
});
