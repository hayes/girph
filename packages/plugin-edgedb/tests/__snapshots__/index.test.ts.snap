// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`edgedb generates schema 1`] = `
"scalar DateTime

type ExplicitEdgeDBUser {
  email: String!
  id: ID!
  name: String
}

type Post {
  author: User!
  content: String
  createdAt: DateTime!
  id: ID!
  published: Boolean!
  title: String!
}

type PostPreview {
  id: ID!
  preview: String
}

type Query {
  me: ExplicitEdgeDBUser
  users: [User!]
}

type User {
  email: String!
  id: ID!
  name: String
  posts: [Post!]!
}"
`;
